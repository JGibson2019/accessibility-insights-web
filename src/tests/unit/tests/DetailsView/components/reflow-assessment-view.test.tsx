// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Assessment } from 'assessments/types/iassessment';
import { AssessmentTestResult } from 'common/assessment/assessment-test-result';
import { AssessmentData } from 'common/types/store-data/assessment-result-data';
import { shallow } from 'enzyme';
import * as React from 'react';
import { Mock } from 'typemoq';

import {
    ReflowAssessmentView,
    ReflowAssessmentViewDeps,
    ReflowAssessmentViewProps,
} from '../../../../../DetailsView/components/reflow-assessment-view';

describe('AssessmentViewTest', () => {
    test('render for requirement', () => {
        const props = generateProps('requirement');
        const rendered = shallow(<ReflowAssessmentView {...props} />);
        expect(rendered.debug()).toMatchSnapshot();
    });

    test('render for gettting started', () => {
        const props = generateProps('getting-started');
        const rendered = shallow(<ReflowAssessmentView {...props} />);
        expect(rendered.debug()).toMatchSnapshot();
    });

    function generateProps(subview: string): ReflowAssessmentViewProps {
        const assessmentMock = Mock.ofType<Assessment>();
        const assessmentDataMock = Mock.ofType<AssessmentData>();
        const assessmentTestResultMock = Mock.ofType<AssessmentTestResult>();

        const reflowProps = {
            deps: {} as ReflowAssessmentViewDeps,
            prevTarget: {},
            currentTarget: {},
            assessmentNavState: {
                selectedTestSubview: subview,
                selectedTestType: assessmentMock.object.visualizationType,
            },
            assessmentData: assessmentDataMock.object,
            assessmentTestResult: assessmentTestResultMock.object,
        } as ReflowAssessmentViewProps;
        return reflowProps;
    }
});
