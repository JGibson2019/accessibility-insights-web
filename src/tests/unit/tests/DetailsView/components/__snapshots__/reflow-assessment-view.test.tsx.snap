// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentViewTest render for gettting started 1`] = `
"<div>
  <TargetChangeDialog deps={{...}} prevTab={{...}} newTab={{...}} />
  <GettingStartedView gettingStartedContent={[undefined]} />
</div>"
`;

exports[`AssessmentViewTest render for requirement 1`] = `""`;
